### PR TITLE
[bitnami/harbor] Sync Harbor chart with upstream

### LINKED ISSUE #6723: [bitnami/harbor] Ingress is wrong for harbor 2.3.0+
**Which chart**:
harbor 10.1.1 (i think older versions are failing too)

**Describe the bug**
Harbor v2.3.0+ is using a new `common.xxxxx.js` file, which is catched by wrong ingres rule:
https://github.com/bitnami/charts/blob/25585c7a042c48ad4b1e8a877c9c3d9b7db6228f/bitnami/harbor/templates/ingress/ingress.yaml#L22
https://github.com/bitnami/charts/blob/25585c7a042c48ad4b1e8a877c9c3d9b7db6228f/bitnami/harbor/templates/ingress/ingress.yaml#L91

**To Reproduce**
Can currently not reproduced, because the new v2.3.0 images of bitnami for harbor are not yet available.

**Expected behavior**
Should work, see goharbor/harbor#14495 and goharbor/harbor-helm#873

**Version of Helm and Kubernetes**: 

- Output of `helm version`: 

```
doesn't matter
```

- Output of `kubectl version`:

```
doesn't matter
```

**Additional context**

### PR DESCRIPTION
**Pull request by joancafom**
Signed-off-by: Jose Antonio Carmona <jcarmona@vmware.com>

**Description of the change**

Recently, Harbor has released a new version [2.3.0](https://github.com/goharbor/harbor/releases/tag/v2.3.0) that includes new features, such as IPv6 support. The aim of this PR is to sync this chart with the latest upstream changes. Below you may find some of the related changes in upstream included in this PR:

_NCP and Ingress rules fixes:_
https://github.com/goharbor/harbor-helm/pull/643/files
https://github.com/goharbor/harbor-helm/commit/42da4e56d19f6d9ba3456b54719dfe75bbfd0023

_ServiceAccount token automount feature:_
https://github.com/goharbor/harbor-helm/pull/886/commits/697d55268aba988c2bcff55172bf76b115fb12f4

_Support for more registry types:_
https://github.com/goharbor/harbor-helm/commit/7c3588a456ca48dffbbf0c7b3a157845bd475ec6

_IPv6 support:_
https://github.com/goharbor/harbor-helm/commit/a871a2212cef8c0d224f497613b1c1ced6f6ec45#diff-503f763af98900b4ec4b0fda4e3ca2a6469e7539ce3bcf6cd961a21ef2a84c6a

**Benefits**

Updates the chart to support the latest changes in upstream, which also include support for the features in the new Harbor version [2.3.0](https://github.com/goharbor/harbor/releases/tag/v2.3.0)

**Possible drawbacks**

Harbor version [2.3.0](https://github.com/goharbor/harbor/releases/tag/v2.3.0) mentions a breaking change in one of its API routes. This is referenced in the README upgrade considerations.

Version 

**Applicable issues**

  - fixes #6723

**Additional information**

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/harbor/templates/registry/registry-cm.yaml [hunk 1]
```diff
@@ -125,7 +125,7 @@ data:
         region: {{ .Values.persistence.imageChartStorage.oss.region }}
         bucket: {{ .Values.persistence.imageChartStorage.oss.bucket }}
         {{- if .Values.persistence.imageChartStorage.oss.endpoint }}
-        endpoint: {{ .Values.persistence.imageChartStorage.oss.endpoint }}
+        endpoint: {{ .Values.persistence.imageChartStorage.oss.bucket }}.{{ .Values.persistence.imageChartStorage.oss.endpoint }}
```
Comment (viceice) [inline bitnami/harbor/templates/registry/registry-cm.yaml:128]: Not sure about this change, does it work with minio as S3 backend ?

**Review by viceice — COMMENTED** [review]
LGTM, just a small thing. 🥰

File: bitnami/harbor/templates/trivy/trivy-sts.yaml [hunk 2]
```diff
@@ -57,7 +57,7 @@ spec:
       {{- if .Values.podSecurityContext }}
       securityContext: {{- include "common.tplvalues.render" (dict "value" .Values.podSecurityContext "context" $) | nindent 8 }}
       {{- end }}
-      automountServiceAccountToken: {{ .Values.trivy.automountServiceAccountToken }}
+      automountServiceAccountToken: {{ .Values.trivy.automountServiceAccountToken | default false }}
```
Comment (miguelaeh) [inline bitnami/harbor/templates/trivy/trivy-sts.yaml:60]: The default should come from the values.yaml so the pipe is not needed. This is repeated in similar lines.

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you very much for the PR!
I just added a picky comment

File: bitnami/harbor/templates/registry/registry-cm.yaml (hunk 1 above)
Comment (joancafom) [reply bitnami/harbor/templates/registry/registry-cm.yaml:128]: For context, this change refers to: https://github.com/goharbor/harbor-helm/pull/675

Regarding your question, minio (or any other `S3` backend) should not be affected by this change as this only applies if the selected storage type for images and charts is `OSS` (and not `S3`) 🙂

**Review by miguelaeh — APPROVED** [review]
LGTM!

**Pull request merged**